README.md

```diff
@@ -1,40 +1,75 @@
+      - [`init` message fields:](#init-message-fields)
+      - [Obtaining Operators data](#obtaining-operators-data)
+  - [Architecture](#architecture)
+    - [Flow Description:](#flow-description)
+    - [Note on DKG instance management](#note-on-dkg-instance-management)
+  - [Security notes](#security-notes)
+  - [Development Notes:](#development-notes)
+    - [TODO:](#todo)
+    - [Additional:](#additional)
+    - [Flow TODO Brakedown](#flow-todo-brakedown)
+      - [Round 1](#round-1)
+      - [Round 2](#round-2)
 
-### Operators data
 
-The data of the operators (ID, IP, Pubkey) can be collected in any way, for example a central server that you can pull the data from, or a preset file where all operators data exist.
+## Goal and Introduction
 
-Information about operators can be collected at `json` file and supplied to initiator to use for a key generation.
+### DKG
 
-Operators info file example (`./examples/operators_integration.json`):
+Distributed Key Generation is a cryptographic process that aims to solve the problem of coordinating `N` parties to cryptographically sign and verify signatures without relying on Trusted Third Parties.
+The process is demonstrated to be successful in successfully computing a key pair in the presence of a number `T` attackers in a decentralized network.
+To do so, this algorithm generates a public key, and a secret key of which no single party knows, but has some share of.
+The involvement of many parties requires Distributed key generation to ensure secrecy in the presence of malicious contributions to the key calculation.
 
-```json
-[
-  {
-    "id": 1,
-    "public_key": "LS0tLS1CRUdJTiBSU0....",
-    "ip": "http://localhost:3030"
-  },
-  {
-    "id": 2,
-    "public_key": "LS0tLS1CRUdJTiB....",
-    "ip": "http://localhost:3031"
-  }
-]
+For more information about DKG in general, [please visit this page](https://en.wikipedia.org/wiki/Distributed_key_generation).
+
+### DKG tool by SSV
+
+The SSV team built this tool leveraging [drand](https://drand.love/)'s DKG protocol implementation ([please visit their documentation](https://drand.love/docs/cryptography/#setup-phase) for more details on it). This implementation operates under the assumption of a p2p network, allowing operators to communicate.
+
+The `ssv-dkg` was built to lift this assumption and provide a communication layer that centered on an Initiator figure, to facilitate communication between operators. The introduced potential risk for centralization and bad actors is handled with signatures and signature verifications, as explained in the [Security notes](#security-notes) section.
+
+Finally, the outcome of the DKG ceremony is a BLS key pair to be used for validator duties by operators on the ssv.network. As such, the tool ends the process by creating a deposit file to activate the newly created validator key pair, and proceeds to generating the payload for the transaction, and triggering the transaction itself.
```

Comment:
we're currently not triggering the tx itself. also I don't know that we're planning to do it.